mountpoint-s3/src/logging/syslog.rs

```diff
@@ -105,7 +106,12 @@ where
     fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
         // No need to do any filtering -- we assume this layer is paired with a filter. So just
         // build the message and ship it.
-        let metadata = event.metadata();
+
+        // Normalize the metadata if it came from a `log` event rather than `tracing`. For us that
```

Comment:
Is it for handling our log messages that had the wrong target and a bunch of extra fields?

Reply:
No, it's for handling log messages that are generated with the [`log` crate](https://crates.io/crates/log). We use [`tracing-log`](https://crates.io/crates/tracing-log) to redirect those log messages to our `tracing` infrastructure, but `tracing-log` adds a bunch of metadata that we don't want. `normalized_metadata` converts that metadata to match a regular `tracing` event.